### PR TITLE
plan, executor: add physical stream agg and remove aggType

### PR DESCRIPTION
**Pull request by hanfei1991**
add stream agg operator
@zimulala

### CODE REVIEW

File: executor/builder.go [hunk 1]
```diff
@@ -611,21 +613,21 @@ func (b *executorBuilder) buildSemiJoin(v *plan.PhysicalHashSemiJoin) *HashSemiJ
...
-			GroupByItems: v.GroupByItems,
-		}
-	}
+func (b *executorBuilder) buildHashAgg(v *plan.PhysicalHashAgg) Executor {
 	return &HashAggExec{
 		baseExecutor: newBaseExecutor(v.Schema(), b.ctx, b.build(v.Children()[0])),
 		sc:           b.ctx.GetSessionVars().StmtCtx,
 		AggFuncs:     v.AggFuncs,
 		GroupByItems: v.GroupByItems,
-		aggType:      v.AggType,
```
Comment (zimulala) [inline executor/builder.go]: Can we remove the field of `aggType`？

File: plan/gen_physical_plans.go [hunk 2]
```diff
@@ -250,13 +253,42 @@ func (p *baseLogicalPlan) generatePhysicalPlans() []PhysicalPlan {
...
+	streamAggs := make([]PhysicalPlan, 0, len(p.possibleProperties))
+	for _, cols := range p.possibleProperties {
+		_, keys := getPermutation(cols, p.groupByCols)
+		if len(keys) != len(p.groupByCols) {
+			continue
+		}
+		agg := basePhysicalAgg{
+			GroupByItems: p.GroupByItems,
+			AggFuncs:     p.AggFuncs,
+		}.initForStream(p.ctx, cols[:len(keys)], p.inputCount)
```
Comment (zimulala) [inline plan/gen_physical_plans.go]: Why not use `keys` directly?

File: executor/builder.go (hunk 1 above)
Comment (hanfei1991) [reply executor/builder.go]: We use different structs for stream and hash agg, so aggType is useless.

**Comment by zimulala** [comment]
LGTM

**Comment by coocood** [comment]
@hanfei1991 
Any test covers it?

File: plan/explain.go [hunk 3]
```diff
@@ -179,10 +179,10 @@ func (p *Limit) ExplainInfo() string {
 }
 
 // ExplainInfo implements PhysicalPlan interface.
-func (p *PhysicalAggregation) ExplainInfo() string {
-	buffer := bytes.NewBufferString(fmt.Sprintf("type:%s", p.AggType))
+func (p *basePhysicalAgg) ExplainInfo() string {
+	buffer := bytes.NewBufferString("")
```
Comment (coocood) [inline plan/explain.go]: Why removes aggregation type?

**Comment by hanfei1991** [comment]
@coocood This is just a refactor. Replace field 'aggType' with an individual operator `PhysicalStreamAggregate`. So no test needs to supply.

**Comment by coocood** [comment]
@hanfei1991 
But we lost some information in explain info.

**Comment by coocood** [comment]
@hanfei1991 
So the title is misleading. Please change it.

File: executor/builder.go [hunk 4]
```diff
@@ -611,21 +613,21 @@ func (b *executorBuilder) buildSemiJoin(v *plan.PhysicalHashSemiJoin) *HashSemiJ
...
+	}
+}
+
+func (b *executorBuilder) buildStreamAgg(v *plan.PhysicalStreamAgg) Executor {
+	return &StreamAggExec{
+		baseExecutor: newBaseExecutor(v.Schema(), b.ctx, b.build(v.Children()[0])),
```
Comment (zz-jason) [inline executor/builder.go]:
```go
childExec := b.build(v.Children()[0])
if b.err != nil {
    b.err = errors.Trace(b.err)
    return nil
}
```

**Comment by hanfei1991** [comment]
@zz-jason PTAL

**Pull request merged**